src/proxy/BaseProxy.ts

```diff
@@ -25,8 +25,8 @@ export class BaseProxy {
         return deferred.promise;
     }
 
-    protected encodeInvokeOnKey<T>(codec: any, key: any, ...codecArguments: any[]): Q.Promise<T> {
-        var partitionId: number = this.client.getPartitionService().getPartitionId(key);
+    protected encodeInvokeOnKey<T>(codec: any, partitionKey: any, ...codecArguments: any[]): Q.Promise<T> {
+        var partitionId: number = this.client.getPartitionService().getPartitionId(partitionKey);
```

Comment:
This does not have to be a key from key-value pair. In non-distributed data structures (List, Set) the structure name is used to determine the partition ID.